New syndie derringer, .38 dart & EMP bullets. Nerfs .38 specialty ammos.

VickiMorris opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
This adds an additional firearm to the Syndicate uplink, the 'Infiltrator' Coat Pistol, a double-barreled derringer chambered in .38-special (The best cartridge in existence) and available for 4 TC (loaded with Match ammo, with an additional speedloader. 2 + 6 shots for your purchase). It is small in size, has "60%" volume per shot (compared to the Stechkin at 80), cannot be silenced, and can be quickloaded using .38 speedloaders.

In addition to this, new .38 "Blisterpack" rounds have been added as a smaller equivalent to 12ga dart slugs. They hold 10 units each (compared to 30u), deal only 4 brute on hit (compared to 6 brute), and similarly do not have any piercing capabilities.

Now introducing .38 "BLK_OUT" rounds, a low damage specialized round that functions as a localized ranged EMP; weaker than most other Syndie EMP gear (Grenades & Implant), and not as stealthy (Flashlight), but allowing ranged applications similar to the ion rifle/carbine.

~~Four~~ SIX ammo purchases are available in the uplink alongside the Coat Pistol; standard, Iceblox, Hot Shot, Blister, BLK_OUT, and Dumdum are all available for 1 TC each (6 shots on a ready-to-use speed loader.)

---

After discussing and reviewing the .38 ammo types available, the damage of Iceblox and Hot Shot rounds has been reduced, and the effect of Hot Shot rounds are heavily reduced. Iceblox still reduces the victim's temperature by 100 degrees, but deals **15** instead of **20** damage. Hot Shot now deals **12** instead of **20** damage, and only applies **2** fire stacks instead of **6** as the later is enough to husk with a single shot.

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.-->
This adds a cheaper "hold-out" style sidearm with a relatively easy to acquire ammo type on the station. The sheer variety of ammo available for the Detective's revolver gives it a lot of utility as a DoT or CC weapon, and having various types of bouncing ammunition can reward the user with landing trickshots. The tweaks to Iceblox and Hot Shot preserve their special qualities while not making them a straight upgrade to standard .38 bullets.

> This also fixes embedding/shrapnel with any projectiles that utilize it, as there was no code that made use of projectile shrapnel actively. Any shrapnel type objects will also be deleted upon removal to prevent someone from throwing a loose bullet/frag grenade shard and having it embed as if it were fired from a gun. If this weapon/other changes are not wanted, I will happily move this fix onto its own PR so it isn't a forced bundle.

This has since been moved to a different PR, to not spread this one too thin --> #9820 

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/0570fcdd-b875-435e-836f-a3ebdf357446)
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/eaf12fa7-7946-48d7-bf15-46e22b80eaff)
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/254a4739-d99e-4298-8b2c-9095d0dad693)

https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/fe4d7d03-8475-4ad0-a44e-a11e84ef4cb3

https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/0d5170eb-3584-4c6e-91f5-e3a718e2b99d

latest revision of Derringer spread, sound, and other tweaks shown in below video (12 MB, didn't want to upload to github). Used alongside Stechkin and Revolver to show comparable sound, damage, and spread while wielded and unwielded. Note, this video does not have the latest upsteam PRs, so Dumdum rounds do not embed in this video as they would in game.

https://cdn.discordapp.com/attachments/1096069498529915012/1153306763483947018/Derringer_Compilation.mp4

</details>

## Changelog
:cl:Impish_Delights
add: Syndicate Coat Pistol; Cheap Sidearm w/ lots of utility focused ammo
add: Coat Pistol, .38 standard, DumDum, HotShot, Iceblox, Blister, and BLK_OUT ammo available in Syndie uplink
add: .38 Blister rounds; 10u dart projectiles, similar to shotgun dart rounds
add: .38 BLK_OUT rounds; low damage localized EMP rounds
imageadd: Added sprite for Coat Pistol & .38 Blister rounds
balance: Iceblox .38 deal less damage (20 -> 15)
balance: Hot Shot .38 deal less damage (20 -> 12) and less fire stacks (6 -> 2)
add: Adds 'Wild Spread' and 'Wild Factor' variables to guns, allows 'minimum' recoil to be set
add: Enables 'Wild Spread' on Coat Pistol only, all other guns unaffected
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

VickiMorris committed:
Added syndie derringer, uplink entries, and .38 dart bullets

VickiMorris committed:
Fix Projectile Embedding, Nerf Hot Shot & Iceblox

Drops Iceblox damage from 20 to 15
Drops Hot Shot damage from 20 to 12
Reduces Hot Shot fire stacks from 6 to 2
Allows projectiles to properly spawn & embed shrapnel
Shrapnel will qdel when removed (so bullets can't be thrown and reembed)

VickiMorris committed:
Added Innate Spread

Derringer has 20 innate spread
Special weapon weight define added (EX_LIGHT) with value of 0
EX_LIGHT weapons only have 20 additional spread when not wielded/two-handed

VickiMorris committed:
Disallows catching of shrapnel

Adds TRUE to skipcatch for hitby proc on shrapnel.
Adds additional comments.

VickiMorris committed:
Removes Embedding Fixes

Removes embedding fixes/tweaks
Changes moved to different PR

VickiMorris committed:
Adds Minimum Spread for Derringer

Adds the wild_spread and wild_factor variables to guns, allowing them to set a minimum spread value instead of zero when calculating spread.
Defaults to FALSE & 0.25 respectively.
Sets wild_spread to TRUE for the derringer, and wild_factor to 0.4

VickiMorris committed:
Tweaks Derringer Spread + Unwield Spread

Modifies Derringer Spread stats once again; Innate Spread = 20 --> 18 | Wild Factor = 0.4 --> 0.7
Bonus spread now has a floor based off of the unwielded spread penalty.
Added debugging variant of the Derringer (12 shot internal magazine)

VickiMorris committed:
Removes Snowflake Weapon Weight define

Removes WEAPON_EX_LIGHT = 0 define from combat.dm
Manually sets spread_unwielded on the derringer to 30 (up from 20)
Spread values now 18 / 48 / 0.7

VickiMorris committed:
Fixes Blistershot Directional Sprites

Blistershot .38 rounds now have green bullet tips from ALL 8 directions, instead of just one.

PowerfulBacon commented on code/modules/uplink/uplink_items.dm:
```diff
@@ -918,6 +926,42 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
+	desc = "A 2-pack of .38-special speed loaders, offering 6 Iceblox and 6 Hot Shot cartridges. Set your \
+	targets alight or freeze them in place with these specialized ammo types."
+	item = /obj/item/storage/box/syndie_kit/c38icyhot
+	cost = 4
```
4 TC seems pretty expensive for these, they aren't *that* good in reality and can be printed from security

VickiMorris replied:
I wasn't aware that hotshot/iceblox was avalible in the sec lathe. I might look into nixxing those for other specialty ammos then. something like the EMP rounds sound neat as a diet Ion gun, but I'd be worried about seeing that get overused by immediately nuking a seccie's baton and disabler. I know that dumdum and match are cargo crate exclusive however.

VickiMorris replied:
Seperated out into each ammo type and made 1 TC.

PowerfulBacon commented on code/modules/uplink/uplink_items.dm:
```diff
@@ -918,6 +926,42 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
+	item = /obj/item/ammo_box/c38/dumdum
+	cost = 2
+	purchasable_from = ~UPLINK_CLOWN_OPS
+	illegal_tech = FALSE
```
These all have big enough variation and different enough roles that it would probably be fine to make them all 1 TC, keeping this gun as the super cheap but kinda crap gun you can buy. All of these ammo types can be printed relatively easilly with the speedloader included unlike other guns which require you to purchase the mags and you can only print the bullets. I think having the more special ammos be cheaper would be good for giving this gun a role of a defensive tool that you can use to slow people down who are onto you, but isn't that good for actual combat

PowerfulBacon replied:
Utility bullets like the ones that nukies can get could make this really good for defensive situation, especially if they could buy bullets that deal less damage but EMP on impact like the sniper rifle gets.

VickiMorris replied:
All ammos reduced to one TC, weak EMP rounds added (BLK_OUT); Dumdum remain as a cargo only purchase (IIRC the PR for embedding that added them didn't make them printable at all) asides from the uplink.

Blister and BLK_OUT remain uplink exclusive.

PowerfulBacon commented on code/modules/projectiles/guns/ballistic/pistol.dm:
```diff
@@ -28,6 +28,32 @@
 	var/obj/item/suppressor/S = new(src)
 	install_suppressor(S)
 
+/obj/item/gun/ballistic/automatic/pistol/der38
```
As a difficult to code but potentially cool suggestion, being able to hide this inside a suit or something so you can sneak it into prison with you would be cool. Of course this is only a suggestion and doesn't need to be completed before merging as it may need us to make the suit attachment thing a component.

VickiMorris replied:
I was debating on making a special set of plain black gloves that can store it, akin to a wrist sling or something. If I wanted to make it attachable to a jumpsuit, I could see about poking around with accessory code and if it's possible to make it not display on examine text.

VickiMorris committed:
.38 EMP Rounds, Cheaper Ammo

Adds .38 'BLK_OUT' cartridges; 8 brute damage, minor EMP effect (0 major, 2 minor range) on impact.
(3, 4 for Sniper EMP | 4, 10 for EMP Grenade | 3, 5 for EMP Implant)
Reduces price of all ammo types to 1 TC in uplink.
Removes 'Thermal Bundle'
Adds BLK_OUT, Iceblox, and Hot Shot to uplink.

VickiMorris committed:
BLK_OUT Round Code Change

Changes their on_hit() to match the on_hit() function of the sniper EMP rounds.

VickiMorris committed:
Unruns your Times

bluh bluh
